src/Illuminate/Auth/Middleware/RequirePassword.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace Illuminate\Auth\Middleware;
+
+use Closure;
+use Illuminate\Routing\Redirector;
+
+class RequirePassword
+{
+    /**
+     * The Redirector instance.
+     *
+     * @var \Illuminate\Routing\Redirector
+     */
+    protected $redirector;
+
+    /**
+     * Create a new middleware instance.
+     *
+     * @param  \Illuminate\Routing\Redirector  $redirector
+     * @return void
+     */
+    public function __construct(Redirector $redirector)
+    {
+        $this->redirector = $redirector;
+    }
+
+    /**
+     * Handle an incoming request.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Closure  $next
+     * @param  string  $redirectToRoute
+     * @return mixed
+     */
+    public function handle($request, Closure $next, $redirectToRoute = null)
+    {
+        if ($this->shouldConfirmPassword($request)) {
+            return $this->redirector->guest(
```

Comment:
@driesvints I think using the `\Illuminate\Contracts\Routing\ResponseFactory` contract and its `redirectGuest` method (which is a wrapper for the redirector guest method) here would be better since that way we could swap the implementation which we can't do here as the redirector doesn't have an interface.

Reply:
And then use the UrlGenerator for the route you mean?

Reply:
Yes.

Reply:
@X-Coder264 feel free to send in a pr with those changes 👍